### PR TITLE
Added support for callback properties (#21)

### PR DESCRIPTION
**Pull request by clauderic**
Hey @kentcdodds, first of all, thanks for this great library, only stumbled upon it a few days ago but already it's truly been a godsend when implementing in-memory search/filtering functionality 😄 

One lacking feature so far however has been support for callback properties. This is especially useful for  interfacing with libraries such as Immutable.js

I've updated the tests and documentation, take a look when you get the chance and let me know if there are any changes you'd like to be made :)

Cheers!

### CODE REVIEW

File: src/index.test.js [hunk 1]
```diff
@@ -129,6 +129,21 @@ const tests = {
...
   },
+  'can handle property callback': {
+    input: [
+      [
+        {name: {first: 'baz'}},
+        {name: {first: 'bat'}},
+        {name: {first: 'foo'}},
+      ],
+      'ba',
+      {keys: [(item) => item.name.first]},
```
Comment (kentcdodds) [inline src/index.test.js]: This is the reason the build is failing. That arrow function doesn't need parentheses.

Comment (clauderic) [inline src/index.test.js]: Fixed!

**Comment by kentcdodds** [comment]
:shipit:

**Comment by kentcdodds** [comment]
Thanks!

**Comment by kentcdodds** [comment]
Thanks so much for your help! I've added you as a collaborator on the project. Please make sure that you review the CONTRIBUTING.md file (specifically the bit about the commit messages and the git hooks) and familiarize yourself with the code of conduct (we're using [the contributor covenant](http://contributor-covenant.org/)). You might also want to [watch the repo](https://help.github.com/articles/watching-repositories/) to be notified when someone files an issue/PR. Please continue to make PRs as you feel the need (you can make your branches directly on the repo rather than your fork if you want). Thanks! And welcome to the team :)

**Comment by clauderic** [comment]
Awesome, thank you! Already have a couple minor improvements in mind 😊

**Pull request merged**